services/provider/client/client.go

```diff
@@ -24,17 +24,33 @@ func NewConsumerClient(cc *grpc.ClientConn, timeout time.Duration) *ConsumerClie
 func NewGRPCConnection(serverAddr string, opts []grpc.DialOption) (*grpc.ClientConn, error) {
 	conn, err := grpc.Dial(serverAddr, opts...)
 	if err != nil {
-		klog.Fatalf("failed to dial: %v", err)
+		return nil, fmt.Errorf("failed to dial: %v", err)
```

Comment:
Probably include this in the previous commit? Unless I'm missing something at this code was not introduced in this PR...

Reply:
This could be in the same PR.  I added the `Initialize` method below and realized that `NewGRPCConnection` is not returning error correctly.  `NewGRPCConnection` is only used by `Initialize` method.

Reply:
This comment is obsolete.  We have only two commits now. One for implementation and other for unit tests.  Marking this as resolved.